src/main/java/de/tum/in/www1/bamboo/server/parser/strategy/PMDParser.java

```diff
@@ -0,0 +1,58 @@
+package de.tum.in.www1.bamboo.server.parser.strategy;
+
+import de.tum.in.www1.bamboo.server.parser.domain.Issue;
+import de.tum.in.www1.bamboo.server.parser.domain.Report;
+import de.tum.in.www1.bamboo.server.parser.domain.StaticCodeAnalysisTool;
+import nu.xom.Document;
+import nu.xom.Element;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PMDParser implements ParserStrategy {
+
+    // XSD for PMD XML reports: https://github.com/pmd/pmd/blob/master/pmd-core/src/main/resources/report_2_0_0.xsd
+    private static final String FILE_TAG = "file";
+    private static final String FILE_ATT_NAME = "name";
+    private static final String VIOLATION_ATT_CLASS = "class";
+    private static final String VIOLATION_ATT_PACKAGE = "package";
+    private static final String VIOLATION_ATT_RULE = "rule";
+    private static final String VIOLATION_ATT_RULESET = "ruleset";
+    private static final String VIOLATION_ATT_PRIORITY = "priority";
+    private static final String VIOLATION_ATT_BEGINLINE = "beginline";
+    private static final String VIOLATION_ATT_ENDLINE = "endline";
+    private static final String VIOLATION_ATT_BEGINCOLUMN = "begincolumn";
+    private static final String VIOLATION_ATT_ENDCOLUMN = "endcolumn";
+
+    @Override
+    public Report parse(Document doc) {
+        Report report = new Report(StaticCodeAnalysisTool.PMD);
+        List<Issue> issues = new ArrayList<>();
+        Element root = doc.getRootElement();
+
+        // Iterate over all <file> elements
+        for (Element fileElement : root.getChildElements(FILE_TAG, root.getNamespaceURI())) {
+            // Extract the file path
+            String filePath = fileElement.getAttributeValue(FILE_ATT_NAME);
+            String shortenedPath = ParserUtils.shortenAndTransformToUnixPath(filePath);
+
+            // Iterate over all <violation> elements
+            for (Element violationElement : fileElement.getChildElements()) {
+                Issue issue = new Issue(shortenedPath);
+
+                issue.setRule(violationElement.getAttributeValue(VIOLATION_ATT_RULE));

```

Comment:
Does it make sense to add a try-catch for every `violationElement` to prevent issues when one child is invalid or do we not need this?

Reply:
The worst case should be that null is returned. For PMD an xsd file exists, which gives guarantees about the structure of document. I think XOM supports checking against that xsd but I don't think this is necessary.